packages/dataviews/src/normalize-fields.ts

```diff
@@ -63,14 +64,24 @@ function getFilterBy< Item >(
 			validOperators.includes( operator )
 		);
 
+		// The `between` operator is not supported when elements are provided.
+		if ( field.elements && operators.includes( OPERATOR_BETWEEN ) ) {
+			operators = operators.filter(
+				( operator ) => operator !== OPERATOR_BETWEEN
+			);
+		}
+
 		// Do not allow mixing single & multiselection operators.
 		// Remove multiselection operators if any of the single selection ones is present.
 		const hasSingleSelectionOperator = operators.some( ( operator ) =>
 			SINGLE_SELECTION_OPERATORS.includes( operator )
 		);
 		if ( hasSingleSelectionOperator ) {
 			operators = operators.filter( ( operator ) =>
-				SINGLE_SELECTION_OPERATORS.includes( operator )
+				// The 'Between' operator is unique as it can be combined with single selection operators.
```

Comment:
As a follow-up, I think we need to re-evaluate this logic that prevents us from having single and multi selection operators registered. I've created ARC-346 to track it.